src/Evolution/Systems/GeneralizedHarmonic/Initialize.hpp

```diff
@@ -0,0 +1,195 @@
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    // First add 3+1 quantities to the box
+    auto _box =
+        db::create_from<db::RemoveTags<>, db::AddSimpleTags<>, compute_tags>(
+            std::move(box));
+
+    // Then compute gauge related quantities
+    const size_t num_grid_points =
+        db::get<::Tags::Mesh<Dim>>(_box).number_of_grid_points();
+
+    // fetch lapse, shift, spatial metric and their derivs through compute tags
+    const auto& lapse = get<gr::Tags::Lapse<DataVector>>(_box);
+    const auto& dt_lapse = get<::Tags::dt<gr::Tags::Lapse<DataVector>>>(_box);
+    const auto& deriv_lapse =
+        get<::Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<Dim>,
+                          Inertial>>(_box);
+    const auto& shift = get<gr::Tags::Shift<Dim, Inertial, DataVector>>(_box);
+    const auto& dt_shift =
+        get<::Tags::dt<gr::Tags::Shift<Dim, Inertial, DataVector>>>(_box);
+    const auto& deriv_shift =
+        get<::Tags::deriv<gr::Tags::Shift<Dim, Inertial, DataVector>,
+                          tmpl::size_t<Dim>, Inertial>>(_box);
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<Dim, Inertial, DataVector>>(_box);
+    const auto& trace_extrinsic_curvature =
+        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(_box);
+    const auto& trace_christoffel_last_indices = get<
+        gr::Tags::TraceSpatialChristoffelFirstKind<Dim, Inertial, DataVector>>(
+        _box);
+
+    // call compute item for the gauge source function
+    using ExtraVars = db::item_type<extras_tag>;
+    ExtraVars extra_vars{num_grid_points};
+    get<GeneralizedHarmonic::Tags::GaugeH<Dim, Inertial>>(extra_vars) =
+        GeneralizedHarmonic::Tags::GaugeHImplicitFrom3p1QuantitiesCompute<
+            Dim, Inertial>::function(lapse, dt_lapse, deriv_lapse, shift,
+                                     dt_shift, deriv_shift, spatial_metric,
+                                     trace_extrinsic_curvature,
+                                     trace_christoffel_last_indices);
+
+    // set time derivatives of GaugeH = 0
+    const auto& dt_gauge_source =
+        make_with_value<tnsr::a<DataVector, Dim, Inertial>>(lapse, 0.);
+
+    // Finally, insert gauge related quantities to the box
+    return std::make_tuple(
+        Initialization::merge_into_databox<
+            InitializeGHAnd3Plus1VariablesTags,
+            // Because DerivCompute<GaugeH> operates on GaugeH wrapped in a
+            // Variables container, we inserted GaugeH that way into the box
+            // (through `extras_tag`), instead of adding `GaugeHCompute`
+            // item here.
+            simple_tags,
+            db::AddComputeTags<
+                ::Tags::DerivCompute<
+                    extras_tag, ::Tags::InverseJacobian<
```

Comment:
We don't see where the two simple tags (GaugeH and dtGaugeH) are updated during the evolution. That is necessary because they are simple tags and not compute tags. They are only simple tags because you need to wrap them in `Variables`.
To remove this hack using the `extras_tag` the `DerivCompute` tag should be changed such that it also accepts tensors as input. This however is something for a separate PR.
Another way of removing this hack would be to write a new compute tag "GaugeHVariablesCompute" that computes a `Variables` instead of a `Tensor`.

```
struct GaugeHVariablesCompute : db::ComputeTag, ::Tags::Variables<tmpl::list<Tags::GaugeH>>
```

Reply:
Hi @hrueter and @nilsleiffischer - agreed, and I have now switched to using the damped harmonic gauge; now both `GaugeH` and its spacetime derivatives are added through compute tags. I had to make some changes to the `DampedHarmonicH` code, which is committed as the first (earliest) commit here. If you like, I can make a separate PR for it (I'd prefer it stay here, but as a separate commit).

Regarding the ugly hack for computing `deriv<GaugeH>`:
 - the right solution is to modify  `DerivCompute` tag to accept tensors as input;
 - therefore, instead of replacing it with yet another - equally hacky - solution, I have removed the need for `deriv<GaugeH>` from this executable. What this means is that it is not observing `3index` and `2index` constraints for now. When `DerivCompute` is properly fixed, it'd be trivial to add them back. In the meanwhile, we have all other constraints being output as well as the deviation of numerical solution from analytic solution - and these measures should be sufficient to diagnose problems / show convergence.